lib/rules/no-restricted-imports.js

```diff
@@ -47,35 +71,77 @@ module.exports = {
 
     create(context) {
         const options = Array.isArray(context.options) ? context.options : [];
-        const isStringArray = typeof options[0] !== "object";
-        const restrictedPaths = new Set(isStringArray ? context.options : options[0].paths || []);
-        const restrictedPatterns = isStringArray ? [] : options[0].patterns || [];
+        const isPathAndPatternsObject =
+            typeof options[0] === "object" &&
+            (options[0].hasOwnProperty("paths") || options[0].hasOwnProperty("patterns"));
+
+        const restrictedPaths = (isPathAndPatternsObject ? options[0].paths : context.options) || [];
+        const restrictedPatterns = (isPathAndPatternsObject ? options[0].patterns : []) || [];
+
+        const restrictedPathMessages = restrictedPaths.reduce((memo, importName) => {
+            if (typeof importName === "string") {
+                memo[importName] = null;
+            } else {
+                memo[importName.name] = importName.message;
+            }
+            return memo;
+        }, {});
 
         // if no imports are restricted we don"t need to check
-        if (restrictedPaths.size === 0 && restrictedPatterns.length === 0) {
+        if (Object.keys(restrictedPaths).length === 0 && restrictedPatterns.length === 0) {
             return {};
         }
 
         const ig = ignore().add(restrictedPatterns);
 
+        /**
+         * Report a restricted path.
+         * @param {node} node representing the restricted path reference
+         * @returns {void}
+         * @private
+         */
+        function reportPath(node) {
+            const importName = node.source.value.trim();
+            const customMessage = restrictedPathMessages[importName];
+            const message = customMessage
```

Comment:
This will cause the default message to get used if an empty string gets used as the custom message, which I'm assuming is unintended. (It's unlikely that users will provide an empty string very often, but I think it would be better to handle this case consistently with the case where non-empty strings are provided.)

Reply:
Could that be resolved by modifying the JSON schema to forbid an empty message?

Reply:
(specifically, by adding `minLength: 1`)

Reply:
That would also work. I don't have any strong preferences for how empty strings are handled; I just want to make sure that the behavior is consistent with the rest of the rule (or that if the behavior is inconsistent, it's a design choice rather than a bug).